### ISSUE TITLE
Missing config files for the MiT + VWFormer experiments

### ISSUE THREAD

**Issue by VanessaBorst** [yan-hao-tian/VW#24]
Hello,

I try to integrate your proposed VW-Head together with the MiT Backbone that is used in Segformer into a custom PyTorch framework for medical segmentation.
I tried to figure out the configuration you used from the supplied MMSegmentation Configs, but unfortunately, the base and also the detailed configs differ for different settings. 

Could be please provide the config files that you used for the experiments in Section 5.2.1 and Table 1?
More specifically, I am interested in the MiT-B3 variant and was wondering, which parameters I need to pass to the VW head for short_cut (True/False) , nheads and in_channels.

Here is what I have figured out up to know:

![Image](https://github.com/user-attachments/assets/fabbce7f-dbbb-4e56-b48d-6904d359f768)

However, with this, I get a total amount of 51.4M trainable parameters (instead of the 47.3M reported in the paper).
I would really appreciate if you could also provide the config files for MiT + VWHead, not only for ConvNeXt and Swin Transformer.

Many thanks in advance

**Comment by yan-hao-tian** [comment]
Thanks for your attention, and sorry for the late reply. I am in my last ph.d year and am somehow busy so have little time checking github.
I will upload segformer-related stuff in two weeks. If you are in a hurry, maybe the 'old-version' branch can help you which presents a very similar model with a little differences.

**Comment by VanessaBorst** [comment]
Thanks for the reply. I already had a look at the 'old-version' branch but did not mange to find the information I search for.
Specifically, I am interested in the combination of MiT-B3 variant and the VWFormer head, not in the traditional Segformer with MLP head.
How should one select  the short_cut and nheads arguement in that case?
Is there any guideline/best practice? 
Unfortunately, I am in a hurry (conference deadline approaching) - so if you could briefly provide some information about this in advance, I would really appreciate it :)

Thanks a lot

**Comment by yan-hao-tian** [comment]
`_base_ = [
    '../_base_/models/segformer_mit-b0.py', 
    '../_base_/datasets/ade20k.py',
    '../_base_/default_runtime.py',
    '../_base_/schedules/schedule_160k.py'
]
crop_size = (512, 512)
checkpoint = 'https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segformer/mit_b3_20220624-13b1141c.pth'  # noqa

# model settings
model = dict(
    pretrained=checkpoint,
    backbone=dict(
        embed_dims=64, num_heads=[1, 2, 5, 8], num_layers=[3, 4, 18, 3]),
    decode_head=dict(
        channels=512,
        in_channels=[64, 128, 320, 512]))

# stage decay learning rate
optimizer = dict(
    _delete_=True,
    type='AdamW',
    lr=0.00003,
    betas=(0.9, 0.999),
    weight_decay=0.01,
    paramwise_cfg=dict(
        custom_keys={
            'pos_block': dict(decay_mult=0.),
            'norm': dict(decay_mult=0.),
            'head': dict(lr_mult=10.)
        }))

lr_config = dict(
    _delete_=True,
    policy='poly',
    warmup='linear',
    warmup_iters=1500,
    warmup_ratio=1e-6,
    power=1.0,
    min_lr=0.0,
    by_epoch=False)

# By default, models are trained on 4 GPUs with 4 images per GPU
data = dict(samples_per_gpu=4)
# fp16 settings
optimizer_config = dict(type='Fp16OptimizerHook', loss_scale='dynamic')
# fp16 placeholder
fp16 = dict()

runner = dict(type='IterBasedRunner', max_iters=160000)
# saving checkpoints every 8000 iters
checkpoint_config = dict(by_epoch=False, interval=8000)
# evaluating every 1000 iters and save the best.
evaluation = dict(interval=1000, metric='mIoU', save_best='mIoU')
`

**Comment by yan-hao-tian** [comment]
# segformer_mit-b0.py
norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained=None,
    backbone=dict(
        type='MixVisionTransformer',
        in_channels=3,
        embed_dims=32,
        num_stages=4,
        num_layers=[2, 2, 2, 2],
        num_heads=[1, 2, 5, 8],
        patch_sizes=[7, 3, 3, 3],
        sr_ratios=[8, 4, 2, 1],
        out_indices=(0, 1, 2, 3),
        mlp_ratio=4,
        qkv_bias=True,
        drop_rate=0.0,
        attn_drop_rate=0.0,
        drop_path_rate=0.1),
    decode_head=dict(
        type='VWHead',
        in_channels=[32, 64, 160, 256],
        in_index=[0, 1, 2, 3],
        channels=512,
        dropout_ratio=0.1,
        num_classes=150,
        short_cut=True,
        nheads=1,
        norm_cfg=norm_cfg,
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))

**Comment by Rudder573** [comment]
> Thanks for your attention, and sorry for the late reply. I am in my last ph.d year and am somehow busy so have little time checking github. I will upload segformer-related stuff in two weeks. If you are in a hurry, maybe the 'old-version' branch can help you which presents a very similar model with a little differences.

Thanks for uploading your project here. Well...I encountered the same question. I adjusted the inputs channels "[32, 64, 160, 256]" for "MiT-B0", but I got the issues that the results are "nan". Futhermore, I'm busy writing the graduation essay and in demand in these results. I would appreciate if you could also provide the config files for MiT + VWHead.

Thanks a lot, and best wishes!
